services/transcribe.py

```
@@ -170,10 +170,26 @@ def _download_with_ytdlp(self, url: str, output_path: Path):
+                except (OSError, PermissionError) as e:
+                    # Clean up the actual file if replace fails
+                    cleanup_error = None
+                    try:
+                        if actual_path.exists():
+                            actual_path.unlink()
+                    except OSError as cleanup_exc:
+                        cleanup_error = cleanup_exc
+                    
+                    # Re-raise with full context including cleanup errors
+                    if cleanup_error:
+                        raise RuntimeError(f"Failed to rename downloaded file from {actual_path} to {output_path}: {e} (cleanup also failed: {cleanup_error})") from e
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)